src/assign.c

```diff
@@ -232,7 +232,7 @@ SEXP setdt_nrows(SEXP x)
       }
       len_xi = INTEGER(dim_xi)[0];
     } else {
-      len_xi = LENGTH(xi);
+      len_xi = length(xi);
```

Comment:
Are any of the other >200 places we use `LENGTH()` worth changing too? (maybe glance through now if anything looks like an obvious fix to put in the patch release, and file a follow-up to go through more carefully in the future)

Reply:
Apparently, we already had this case before item 3 of `1.7.2`

> A crash bug is fixed when j returns a (strictly) NULL column next
        to a non-empty column, #1633. This case was anticipated and coded
        for but an errant LENGTH() should have been length(). Thanks
        to Dennis Murphy for reporting.

Might be worth to check carefully

Reply:
I filed #6877 for this